zingolib/src/wallet/tx_map/spending_data.rs

```diff
@@ -0,0 +1,20 @@
+//! the subsection of TxMap that only applies to spending wallets
+
+use crate::data::witness_trees::WitnessTrees;
+
+/// the subsection of TxMap that only applies to spending wallets
+pub(crate) struct SpendingData {
+    witness_trees: WitnessTrees,
+}
+
+impl SpendingData {
```

Comment:
Why add an additional layer of type wrapper? Why disambiguate between a WitnessTrees, and a SpendingData?